tests/ga/test_remoteaccess_handler.py

```diff
@@ -0,0 +1,358 @@
+        self.assertEqual(1, mockOSUtil.all_users.keys().__len__())
+        self.assertTrue(testusr in mockOSUtil.all_users, "Expected user {0} missing".format(testusr))
+
+    @patch('azurelinuxagent.common.utils.cryptutil.CryptUtil.decrypt_secret',
+    return_value="]aPPEv}uNg1FPnl?")
+    def test_handle_new_user(self, _):
+        rah = RemoteAccessHandler()
+        rah.os_util = MockOSUtil()
+        data_str = load_data('wire/remote_access_single_account.xml')
+        remote_access = RemoteAccess(data_str)
+        tstuser = remote_access.user_list.users[0].name
+        expiration_date = datetime.utcnow() + timedelta(days=1)
+        expiration = (expiration_date).strftime("%a, %d %b %Y %H:%M:%S ") + "UTC"
+        remote_access.user_list.users[0].expiration = expiration
+        rah.remote_access = remote_access
+        rah.handle_remote_access()
+        os_util = rah.os_util
+        os_util.__class__ = MockOSUtil
+        self.assertTrue(tstuser, "{0} missing from users".format(tstuser))
+        actual_user = os_util.all_users[tstuser]
+        self.assertEqual(actual_user[7], (expiration_date + timedelta(days=1)).strftime("%Y-%m-%d"))
+        self.assertEqual(actual_user[4], "JIT Account")
+
+    def test_do_not_add_expired_user(self):
+        rah = RemoteAccessHandler()
+        rah.os_util = MockOSUtil()      
+        data_str = load_data('wire/remote_access_single_account.xml')
+        remote_access = RemoteAccess(data_str)
+        expiration = (datetime.utcnow() - timedelta(days=2)).strftime("%a, %d %b %Y %H:%M:%S ") + "UTC"
+        remote_access.user_list.users[0].expiration = expiration
+        rah.remote_access = remote_access
+        rah.handle_remote_access()
+        os_util = rah.os_util
+        os_util.__class__ = MockOSUtil
+        self.assertFalse("testAccount" in os_util.all_users)
+
+    def test_error_add_user(self):
+        rah = RemoteAccessHandler()
+        rah.os_util = MockOSUtil()
+        tstuser = "foobar"
+        expiration = datetime.utcnow() + timedelta(days=1)
+        pwd = "bad password"
+        rah.add_user(tstuser, pwd, expiration)
+        os_util = rah.os_util
+        os_util.__class__ = MockOSUtil
+        self.assertFalse(tstuser in os_util.all_users)  
+
+    def test_handle_remote_access_no_users(self):
+        rah = RemoteAccessHandler()
+        rah.os_util = MockOSUtil()
+        data_str = load_data('wire/remote_access_no_accounts.xml')
+        remote_access = RemoteAccess(data_str)
+        rah.remote_access = remote_access
+        rah.handle_remote_access()
+        os_util = rah.os_util
+        os_util.__class__ = MockOSUtil
+        self.assertEqual(0, os_util.all_users.keys().__len__())
+
+    def test_handle_remote_access_validate_jit_user_valid(self):
+        rah = RemoteAccessHandler()
+        result = rah.validate_jit_user("JIT Account")
+        self.assertTrue(result, "JIT account incorrectly identified.")
```

Comment:
The error text (displayed if the test fails) isn't specific, and it's identical to the three following assertFalse() tests, which seems wrong.

Reply:
clarified.

---
In reply to: [194540072](https://github.com/Azure/WALinuxAgent/pull/1171#discussion_r194540072) [](ancestors = 194540072)